### PR TITLE
add service tag to container

### PR DESCRIPTION
**Pull request by kangyili**
### What does this PR do?

Add ecs task service tag to container in ECS env.

The service tag is only available in the core agent when the feature flag `ecs_task_collection_enabled` is enabled. https://github.com/DataDog/datadog-agent/blob/30240ce8183d49b0fb1317e09c9a341a2d6a8999/comp/core/workloadmeta/collectors/util/ecs_util.go#L31 This PR adds the service tag in the tagger and allows the process agent to use the remote tagger to retrieve the new tag.

### CODE REVIEW

File: cmd/process-agent/command/main_common.go [hunk 1]
```diff
@@ -188,7 +188,10 @@ func runApp(ctx context.Context, globalParams *GlobalParams) error {
 
 		// Provide the corresponding tagger Params to configure the tagger
 		fx.Provide(func(c config.Component) tagger.Params {
-			if c.GetBool("process_config.remote_tagger") {
+			if c.GetBool("process_config.remote_tagger") ||
+				// If the agent is running in ECS or ECS Fargate and the ECS task collection is enabled, use the remote tagger
+				// as remote tagger can return more tags than the local tagger.
+				((ddconfig.IsECS() || ddconfig.IsECSFargate()) && c.GetBool("ecs_task_collection_enabled")) {
 				return tagger.NewNodeRemoteTaggerParams()
```
Comment (kangyili) [inline cmd/process-agent/command/main_common.go:195]: remote tagger will be used here https://github.com/DataDog/datadog-agent/blob/7cac8550a612f31d31b3124b27ff79fa3075aa53/pkg/process/util/containers/containers.go#L123

I assume that the tags from the local tagger and the remote tagger should be identical, except for this new service tag

File: comp/core/tagger/taggerimpl/collectors/workloadmeta_extract.go [hunk 2]
```diff
@@ -465,6 +465,10 @@ func (c *WorkloadMetaCollector) handleECSTask(ev workloadmeta.Event) []*types.Ta
 		addResourceTags(taskTags, task.Tags)
 	}
 
+	if task.ServiceName != "" {
+		taskTags.AddLow(tags.EcsServiceName, strings.ToLower(task.ServiceName))
```
Comment (robertjli) [inline comp/core/tagger/taggerimpl/collectors/workloadmeta_extract.go:469]: Could be good to add a test for this?

Comment (davidor) [reply comp/core/tagger/taggerimpl/collectors/workloadmeta_extract.go:469]: Please add a test for it as @robertjli suggested 🙏 

I recently added some tests for tags that were not being tested. Maybe these PRs can be useful as an example: https://github.com/DataDog/datadog-agent/pull/26823 https://github.com/DataDog/datadog-agent/pull/26822

Comment (kangyili) [reply comp/core/tagger/taggerimpl/collectors/workloadmeta_extract.go:469]: thanks for the example ! https://github.com/DataDog/datadog-agent/pull/28660/commits/ee795885441008e2e1210b872d34dafcb749dbf4

**Review by davidor — APPROVED** [review]

**Review by daniel-taf — APPROVED** [review]

**Pull request merged**